cloud/amazon/iam_saml_federation.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: iam_saml_federation
+version_added: "2.3"
+short_description: maintain iam saml federation configuration.
+description:
+    - Maintains IAM SAML identity federation configurations.  This module has a dependency on python-boto >= 2.5.
+options:
+  name:
+    description:
+      - The name of the provider to create.
+    required: true
+  saml_metadata_document:
+   description:
+      - An  XML  document  generated by an identity provider (IdP) that supports SAML 2.0.
+  state:
+    description:
+      - Whether to create or delete identity provider.  If 'present' is specified it will attempt to update the identity provider matching the name field.
+    default: present
+    choices: [ "present", "absent" ]
+extends_documentation_fragment:
+ - aws
+author: "Aidan Rowe (@aidan-)"
+'''
+
+EXAMPLES = '''
+# Note: None of these examples set aws_access_key or aws_secret_key.
+# It is assumed that their matching environment variables are set.
+
+# Creates a new iam saml identity provider if not present
+- name: saml provider
+  iam_saml_federation:
+    name: example1
+    saml_metadata_document: >
```

Comment:
Can you also put a comment in here about what the `>` means when opening a text block? Something to the effect of:
```
# the > here opens an indented block, so no escaping/quoting is needed when in the indentation level under this key
```